crates/compiler/can/src/suffixed.rs

```diff
@@ -828,3 +834,25 @@ fn is_expr_task_ok<'a>(loc_expr: &'a Loc<Expr<'a>>) -> bool {
         _ => false,
     }
 }
+
+/// Detect when we are applying an identical Pattern and Ident
+/// e.g. `Task.await foo \#a!0 -> #a!0` is the same as `foo`
+fn is_intermediate_answer_with_same_ident<'a>(
```

Comment:
@rtfeldman 

If this statement isn't correct, and it should be `Task.await foo \#a!0 -> Task.ok #a!0` instead, then I need to also detect and apply `Task.ok` at suitable locations, and then also handle the `Var` within the `Task.ok` here.

Reply:
Ok, so after sleeping on this and looking at the docs closer `Task.await : Task a b, (a -> Task c b) -> Task c b` and with my pencil and paper handy, I'm pretty confident it should be the latter with `Task.ok #!a0`. 

I'll update for that and fix clips too.